cmd/pitr/collector/collector.go

```diff
@@ -242,35 +242,53 @@ func (c *Collector) CollectBinLogs(ctx context.Context) error {
 	if err != nil {
 		return errors.Wrap(err, "get GTID sets")
 	}
-	var sourceID string
-	for i := len(list) - 1; i >= 0 && sourceID == ""; i-- {
-		sourceID = strings.Split(list[i].GTIDSet, ":")[0]
+	var lastGTIDSetList []string
+	for i := len(list) - 1; i >= 0 && len(lastGTIDSetList) == 0; i-- {
+		gtidSetList := list[i].GTIDSet.List()
+		if gtidSetList == nil {
+			continue
+		}
+		lastGTIDSetList = gtidSetList
 	}
 
-	if sourceID == "" {
+	if len(lastGTIDSetList) == 0 {
 		log.Println("No binlogs to upload")
 		return nil
 	}
 
-	c.lastSet, err = c.lastGTIDSet(ctx, sourceID)
-	if err != nil {
-		return errors.Wrap(err, "get last uploaded gtid set")
+	for _, gtidSet := range lastGTIDSetList {
+		sourceID := strings.Split(gtidSet, ":")[0]
+		c.lastUploadedSet, err = c.lastGTIDSet(ctx, sourceID)
+		if err != nil {
+			return errors.Wrap(err, "get last uploaded gtid set")
+		}
+		if !c.lastUploadedSet.IsEmpty() {
+			break
+		}
 	}
 
 	lastUploadedBinlogName := ""
 
-	if c.lastSet != "" {
-		for i := len(list) - 1; i >= 0; i-- {
-			if list[i].GTIDSet == c.lastSet {
-				lastUploadedBinlogName = list[i].Name
-				break
+	if !c.lastUploadedSet.IsEmpty() {
+		for i := len(list) - 1; i >= 0 && lastUploadedBinlogName == ""; i-- {
+			for _, gtidSet := range list[i].GTIDSet.List() {
+				for _, lastUploaded := range c.lastUploadedSet.List() {
+					isSubset, err := c.db.GTIDSubset(ctx, lastUploaded, gtidSet)
```

Comment:
last uploaded GTID set can be superset of the current GTID set too. 

for example, collector was collecting from pod-0. pod-0 has the following binlogs:
```
binlog.000011 fbb06ce1-1579-11ee-902a-774be528cc87:1-352
binlog.000012 fbb06ce1-1579-11ee-902a-774be528cc87:353-553
binlog.000013 fbb06ce1-1579-11ee-902a-774be528cc87:554-754
binlog.000014 fbb06ce1-1579-11ee-902a-774be528cc87:755-957
binlog.000015 fbb06ce1-1579-11ee-902a-774be528cc87:958-1087
binlog.000016 fbb06ce1-1579-11ee-902a-774be528cc87:1088-1116
```

then I killed pod-0 and collector connected to pod-1 and uploaded the following binlog:
```
binlog.000014 fbb06ce1-1579-11ee-902a-774be528cc87:1-1185
```

after pod-0 comes back and collector connects to it, it checks if `binlog.000014 fbb06ce1-1579-11ee-902a-774be528cc87:1-1185` (last uploaded) is a subset of GTIDs in pod-0 but it's not, it's a superset.

Reply:
https://github.com/percona/percona-xtradb-cluster-operator/pull/1441/commits/8ce508d0bc168996f55fd5939dadab49bdf1b2e0